### DISCUSSION TITLE
ANSWER: Detailed Install Instructions

### DISCUSSION THREAD

**Discussion by brianhughesspe** [mtalcott/google-photos-deduper#46] [General]
Not sure if all of this is absolutely necessary as I too am a novice coder, but this is what worked for me:

**Setup**
1. Install [Docker Desktop](https://docs.docker.com/desktop/) on your system.
Install [GitHub Desktop](https://desktop.github.com/) on your system.
Install [Visual Studio Code](https://code.visualstudio.com/download) on your system.
Install [Python](https://www.python.org/downloads/) on your system

2. From GitHub Desktop, go to File>Clone a repository>URL
a. Add https://github.com/mtalcott/google-photos-deduper.git to the URL field, and choose the path you want it installed to (this should download the files you need to your machine, which we will need for step 4)

3. Create a Google Cloud project and OAuth credentials.
  - Create a Google Cloud project ([Guide](https://developers.google.com/workspace/guides/create-project))
    - Project name: Enter Photos Deduper
    - Select the project
  - Go to APIs & Services > Enable APIs and Services
    - Search for Photos Library API
    - Enable
  - Go to APIs & Services > OAuth consent screen
    - User Type: Choose External
    - Create
      - App name: Enter Photos Deduper
      - User support email: Choose your email
      - Developer contact information: Enter your email
      - Save and Continue
    - Add or remove scopes:
      - Manually add scopes:
        - https://www.googleapis.com/auth/userinfo.profile
        - https://www.googleapis.com/auth/userinfo.email
        - https://www.googleapis.com/auth/photoslibrary
      - Update
      - Save and Continue
    - Test users:
      - Add your email (and any others you want to use the tool with)
      - Save and Continue
  - Go to APIs & Services > Credentials > Create Credentials > OAuth client ID
    - Application type: Choose Web application
    - Name: Enter Photos Deduper Web Client
    - Authorized JavaScript origins: Enter http://localhost/
    - Authorized redirect URIs: Enter http://localhost/auth/google/callback
    - Create
  - Download the JSON file

4. Set up local environment variables.
  - From GitHub Desktop, choose to Open the repository in your external editor
![Untitled1](https://github.com/mtalcott/google-photos-deduper/assets/34150725/e7883f5e-024d-4ec3-8f05-bea8f60fa0d5)

  - From the items on the left, right click example.env, copy, and paste into the same location, rename that new file as simply “.env” without the quotes
![Untitled2](https://github.com/mtalcott/google-photos-deduper/assets/34150725/f8b7edcf-b0e0-4f63-8e3a-c0f64f854bc2)

  - Open terminal/command prompt and Generate [FLASK_SECRET_KEY](https://flask.palletsprojects.com/en/2.3.x/config/#SECRET_KEY) with: 
    - python3 -c 'import secrets; print(secrets.token_hex())' 
    - Copy the output and add it to .env file from 4.b. Replacing the XYZ
  - Add GOOGLE_CLIENT_ID and GOOGLE_CLIENT_SECRET from the client_id and client_secret values from the client secret file created above (JSON that you downloaded in step 3.e.
    - Add it to .env file from 4.b. Replacing the 123-ABC.apps.googleusercontent.com and ImAs3cREt!1, respectively
  - File>save

**Start**

1. Run docker-compose up from the project directory.
  - Easiest way for me to do this was to open the repository location in finder/file explorer and opening a new terminal/command prompt starting at that folder
![Untitled4](https://github.com/mtalcott/google-photos-deduper/assets/34150725/053ac8a4-ceaa-4643-9ff7-9fd94fa94116)

  - Then from terminal/cmd, Run docker-compose up 
2. Load http://localhost/ and follow the instructions from there!
3. [Install the Chrome Extension](https://github.com/mtalcott/google-photos-deduper/blob/main/chrome_extension/README.md) once you want to delete duplicates.

**Comment by aleon1220** [comment]
thanks for the detailed instructions. You should add a readme page Pull Request with instructions for mac.
I did run everything from Linux Ubuntu 22 with no issue as everything was rather native and straight forward...

**Reply by jayluke7896** [reply]
I am unsure of the following: 

Open terminal/command prompt and Generate [FLASK_SECRET_KEY](https://flask.palletsprojects.com/en/2.3.x/config/#SECRET_KEY) with:

python3 -c 'import secrets; print(secrets.token_hex())'
Copy the output and add it to .env file from 4.b. Replacing the XYZ
Add GOOGLE_CLIENT_ID and GOOGLE_CLIENT_SECRET from the client_id and client_secret values from the client secret file created above (JSON that you downloaded in step 3.e.

Add it to .env file from 4.b. Replacing the 123-ABC.apps.googleusercontent.com and ImAs3cREt!1, respectively
File>save

Please DETAIL like I am a four year old sadly due to my lack of coding skills.

Thanks heaps, Jaymie

**Reply by aleon1220** [reply]
@jayluke7896 is not going to work anymore due to changes in the Google API. You may still try but i would suggest using either Mac or a linux distribution. Please keep in mind that my linux laptop was already set up fully with docker and my browser was connected to my google account. I follow the instructions once and it got stuck in the middle as i had many duplicates. I re-executed the batch process and it did perform an initial clean up.

refer to the links and all the best.
https://developers.google.com/photos/support/api-policy 
https://developers.google.com/photos/support/updates

**Comment by Knockknockxx** [comment]
In Visual Studio Code,...(external editor) /Explorer. I've got no items on the left? Am I missing something? Thank you

**Comment by WimHWimH** [comment]
I would like to try this but I'm stuck here : Download the JSON file
Where do I download what Json file??? And what do I do with it afterwards???
Thanks! 

Wim

**Reply by AboNady** [reply]
as mentioned here
Go to APIs & Services > Credentials > Create Credentials > OAuth client ID
Application type: Choose Web application
Name: Enter Photos Deduper Web Client
Authorized JavaScript origins: Enter http://localhost/
Authorized redirect URIs: Enter http://localhost/auth/google/callback
Create
Download the JSON file
![dfgdfd](https://github.com/user-attachments/assets/ececd7b4-55c7-4f7f-b299-ef366dfed7c7)
Then change the keys according to what you get, and rename the example-env file to *.env* only
